### PR TITLE
Adjust TokenMiddleware, and change api/middleware.go name to templatize 

### PR DESCRIPTION
**Pull request by damnski**
* correct variable order for bcrypt.CompareHashAndPassword in func TokenMiddleware
* append tmpl to api/middleware.go and it_s test - so the files are templated when splat_ted.

### CODE REVIEW

File: api/middleware.go.tmpl [hunk 1]
```diff
@@ -52,8 +52,9 @@ func TokenMiddleware(psk []byte, public map[string]string, h http.Handler) http.
 			log.Infof("Authenticating token for protected URL '%s'", r.URL)
 
 			htoken := r.Header.Get("X-Auth-Token")
-			if err := bcrypt.CompareHashAndPassword([]byte(htoken), psk); err != nil {
-				log.Warnf("Unable to authenticate session for '%s' with '%s'", r.URL, htoken)
+			if err := bcrypt.CompareHashAndPassword(psk, []byte(htoken)); err != nil {
```
Comment (fishnix) [inline api/middleware.go.tmpl:55]: I think this is backwards:

https://godoc.org/golang.org/x/crypto/bcrypt#CompareHashAndPassword

`func CompareHashAndPassword(hashedPassword, password []byte) error`

The hashed password is what comes in via the header.  The plaintext password is what's in the configuration for the API.

File: api/middleware.go.tmpl [hunk 2]
```diff
@@ -52,8 +52,9 @@ func TokenMiddleware(psk []byte, public map[string]string, h http.Handler) http.
...
+				log.Warnf("Unable to authenticate session for url: '%s', with error: '%s'", r.URL, err)
+				log.Debugf("Unable to authenticate session for url: '%s', with error: '%s', with user input: '%s'", r.URL, err, htoken)
```
Comment (fishnix) [inline api/middleware.go.tmpl:57]: I think you can add this to an API if you're debugging something.  It doesn't need to be in the middleware.

Comment (fishnix) [reply api/middleware.go.tmpl:57]: if you want to just change the warning message to have the error and not the `htoken`, that's probably better although bcrypt is a one way hash, so theoretically it should be fairly safe to include it.

File: api/middleware.go.tmpl (hunk 1 above)
Comment (damnski) [reply api/middleware.go.tmpl:55]: That explains it.  I think we could probably document the direction better where the bcrypt is supposed to occur - in the config file on the daemon or on the client - unless I missed that or misread it.

Comment (damnski) [reply api/middleware.go.tmpl:55]: I was holding the uuid bcrypted in the config file of the daemon lol.

Comment (fishnix) [reply api/middleware.go.tmpl:55]: ah - that didn't change.  the config file remains the same as it was before.

File: api/middleware.go.tmpl (hunk 2 above)
Comment (damnski) [reply api/middleware.go.tmpl]: yeah, I guess it's well enough understood in the middleware as-is.  I did drop the htoken.  A log.Debugf statement can be added by the implementer of the template as you stated to debug inputs, etc.

**Pull request merged**